### ISSUE TITLE
Adding new connectivity indexes

### ISSUE THREAD

**Issue by cmista** [mne-tools/mne-connectivity#50]
#### Describe the new feature or enhancement
The simplest way to check the relation between two signals is using correlation or a similar a similar approach. However, there are connectivity indexes that don't need to make the assumption of linearity between the interaction of the signals. These kind of indexes can be developed based on the information theory.  

#### Describe your proposed implementation
New functions in the Connectivity Estimation group would be nice to be incorporated. 

#### Describe possible alternatives
Mutual information is a good option as a non-directed measure.
Transfer entropy seems to be a good option for a directed measure (causal) - although this most likely is too complex to handle myself alone  

#### Additional comments
A review about connectivity indexes
Bastos André M., Schoffelen Jan-Mathijs. A Tutorial Review of Functional Connectivity Analysis Methods and Their Interpretational Pitfalls. Frontiers in Systems Neuroscience  

Transfer entropy paper
Thomas Schreiber. Measuring Information Transfer. https://arxiv.org/abs/nlin/0001042v1

A nice article about information theory in neuroscience
Nicholas M. Timme and Christopher Lapish. A Tutorial for Information Theory in Neuroscience. Eneuro

**Comment by jshanna100** [comment]
Yes, very much, and it's been on my mind since the summer, but have been very busy lately. Realistically I think I could find some time for it in December.

**Comment by adam2392** [comment]
Hi @brovelli thanks for pointing those out! I'll definitely have to take a look at the implemented FC measures. Perhaps even port them over here if it makes sense? 

For Frites, what is the general direction of the repo? Are you mainly interested in implementing more connectivity measures, or also interested in consolidating the API?

We have mainly been focused on abstractions of connectivity data structures that cleanly and seamlessly carry over from MNE-Python Raw/Epochs objects.

**Comment by brovelli** [comment]
Hi @adam2392,

I think @EtienneCmb can answer you more precisely concerning the future of the API of Frites and data structures. I don't think they will evolve much.

For what concerns complementary FC measures, indeed together with @EtienneCmb and @ViniciusLima94 , we are currently working on single trial estimates. We are particularly interested in single-trial FC measures, because they allow information theoretical with behavioral variables as a second step.

We are currently testing single-trial time-frequency coherence. As soon as we push it to Frites, we will let you know (decembre?).

Concerning your questions about porting such measures into MNE-connectivity from Frites, franckly I don't know what would be the best strategy. Problems with porting them may arises when the FC measures are not completely finalised and fully tested, and incompatibilities may occur between Frites and MNE-connectivity, if we keep developping them on our side... but again, I am not the best person to ask about this issue. 

Fully open to coordinate and share our efforts between MNE-connectivity and FRITES.

What is your view on this? @agramfort

**Comment by agramfort** [comment]
I think that any collaborative effort is great. Alone you go fast but
together we go far !

however I really don't have the bandwidth to coordinate collaboration here.

I suggest you guys plan a short call to discuss your visions and maybe a
plan
a short code sprint (even remote) to see how far you can go together

>

**Comment by brovelli** [comment]
Ok, thanks Alex. I can try to coordinate this. Indeed a Zoom meeting would help.

So far: 
1) @brovelli 
2) @adam2392 
3) @EtienneCmb 
4) @ViniciusLima94

Who else woudl like to join?

**Comment by jshanna100** [comment]
I would, as well.

**Comment by adam2392** [comment]
That sounds good to me

**Comment by adam2392** [comment]
@brovelli just wanted to follow up to see if perhaps we might want to schedule something by next week (before Thanksgiving for us USA folks :p)?

My email is adam2392 at gmail dot com so feel free to email me as well / send me a gcal invite.
